app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala

```diff
@@ -182,24 +187,43 @@ class AMLSController @Inject()(
                 Ok(amlsDetailsTemplate(form, amlsBodies))
               },
               validForm => {
-                val supervisoryBodyData =
-                  amlsBodies.getOrElse(validForm.amlsCode, throw new Exception("Invalid AMLS code"))
-                val continue = toTaskListOrCheckYourAnswers(isChanging)
-                updateAmlsJourneyRecord(
-                  agent,
-                  amlsData =>
-                    Some(amlsData.copy(
-                      amlsDetails = Some(AmlsDetails(supervisoryBodyData, Right(RegisteredDetails(validForm.membershipNumber, validForm.expiry))))))
-                ).map(
-                  _ => Redirect(continueOrStop(continue, routes.AMLSController.showAmlsDetailsForm()))
-                )
+                amlsService.validateAmlsSubscription(validForm).flatMap {
+                  case AmlsSuspended | _: AmlsCheckFailed => Redirect(routes.AMLSController.showAmlsRecordIneligibleStatus)
+                  case DateNotMatched | RecordNotFound    => Redirect(routes.AMLSController.showAmlsDetailsNotFound)
+                  case ResultOK => {
+                    val supervisoryBodyData =
+                      amlsBodies.getOrElse(validForm.amlsCode, throw new Exception("Invalid AMLS code"))
```

Comment:
What does the user get if this exception is thrown?

Reply:
an general error (Sorry, there is a problem with the service) but the form validator would have prevented a amls body that is not on the list.